vm/systemSmartContracts/delegation.go

```diff
@@ -583,7 +580,7 @@ func (d *delegation) stakeNodes(args *vmcommon.ContractCallInput) vmcommon.Retur
 
 	successKeys, _ := getSuccessAndUnSuccessKeys(vmOutput.ReturnData, args.Arguments)
 	for _, successKey := range successKeys {
-		moveNodeFromList(dStatus.NotStakedKeys, dStatus.StakedKeys, successKey)
+		dStatus.NotStakedKeys, dStatus.StakedKeys = moveNodeFromList(dStatus.NotStakedKeys, dStatus.StakedKeys, successKey)
```

Comment:
I think you should add in this for another line like this one:
dStatus.UnStakedKeys, dStatus.StakedKeys = moveNodeFromList(dStatus.UnStakedKeys, dStatus.StakedKeys, successKey)

Reply:
done